commodore/refs.py

```diff
@@ -136,11 +140,14 @@ def _find_refs(self, prefix, params):
             # Otherwise, handle as leaf
             self._find_ref(prefix, params)
 
-    def find_refs(self):
+    def find_refs(self, target: str, component: str):
         """
-        Search for Kapitan secret refs in `self.parameters`
+        Search for Kapitan secret refs for key `component` (dashes replaced
+        with underscores) in key `parameters` for Kapitan target `target`.
         """
-        self._find_refs("", self.parameters)
+        component_key = component.replace("-", "_")
```

Comment:
This only works for actual components. If `find_refs()` is called on another key in `parameters` (not an alias/component) this replacement is not valid:
For example in the following, the key `mariadb-xlarge` is used:

```
parameters:
  mariadb-xlarge:
    uuid: f342d86a-2b19-11eb-adc1-0242ac120002
```

Here the compilation fails:

```
commodore catalog compile c-bold-water-681

[...]

Traceback (most recent call last):
  File "/usr/local/bin/commodore", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/commodore/cli.py", line 260, in main
    commodore.main(prog_name="commodore", auto_envvar_prefix="COMMODORE")
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/decorators.py", line 73, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/commodore/cli.py", line 132, in compile_catalog
    _compile(config, cluster)
  File "/usr/local/lib/python3.8/site-packages/commodore/compile.py", line 154, in compile
    update_refs(config, aliases, kapitan_inventory)
  File "/usr/local/lib/python3.8/site-packages/commodore/refs.py", line 189, in update_refs
    rb.find_refs(bootstrap_target, i)
  File "/usr/local/lib/python3.8/site-packages/commodore/refs.py", line 149, in find_refs
    params = self.inventory[target]["parameters"][component_key]
KeyError: 'mariadb_xlarge'
```

You can use cluster `c-bold-water-681` to test this.

Reply:
Thanks for pointing this out. The replacement was intended for component keys only, but got moved into `find_refs()` at some point when I was refactoring the code after fixing the immediate issue yesterday.

Reply:
Fixed in the latest version and updated test to have a non-component section which has dashes in the key.

Reply:
Nice, now it works as expected 👍